Currency-Fragments:

LeonidVolohov opened the pull request:
Migrate currency from activity to fragment #20

kramarov-evg committed:
Code clean-up

Changes:
- Tiny fixes to code style

kramarov-evg committed:
Created base for API communication

Changes:
- Defined data model
- Created API-communicator interface

kramarov-evg committed:
Created functionality for API communication

Changes:
- Created functions to retrieve data (w/caching)

LeonidVolohov committed:
Currency-Fragments:
 - Added extentions to build.gradle file for directly handling for UI component as global field
 - Added CurrencyFragment.kt and fragment_currency.xml files for BottomNavigationView

kramarov-evg committed:
Non-working Commit

LeonidVolohov committed:
Merge branch 'stocks_api' of https://github.com/LeonidVolohov/Stonks into currency-fragments

 Conflicts:
	app/build.gradle

LeonidVolohov committed:
Currency-Fragments:
 - Small changes in fragment_currency.xml
 - Changed default fragment in BottomNavigationView

LeonidVolohov committed:
Currency-Fragments:
 - fragment_currency.xml:
     - Added button toggle group for chart length period
     - Changed USD icon to icon of European Union
 - Changed type of input value of getLineData in StockLineChart.kt from List<String> to List<Double>

LeonidVolohov committed:
Currency-Fragments-API:
 - Moved all data classes into one main CurrencyDataModel.kt and removed unnecessary fields which were not used
 - Created CurrencyApiDataUtils.kt for response data

LeonidVolohov committed:
Currency-Fragments-API:
 - Created CurrencyFragmentUtils.kt for all CurrencyFragment.kt API logic
 - Removed in fragment_currency.xml third spinner for chart length and added ButtonToggleGroup instead of it

LeonidVolohov committed:
Currency-Fragments-API:
 - Created CurrencyFragmentUtils.kt for all CurrencyFragment.kt API logic
 - Removed in fragment_currency.xml third spinner for chart length and added ButtonToggleGroup instead of it

LeonidVolohov committed:
Merge remote-tracking branch 'origin/currency-fragments' into currency-fragments

# Conflicts:
#	app/src/main/java/com/stonks/ui/currency/CurrencyFragmentUtils.kt

LeonidVolohov committed:
Merge remote-tracking branch 'origin/currency-fragments' into currency-fragments

# Conflicts:
#	app/src/main/java/com/stonks/ui/currency/CurrencyFragmentUtils.kt

LeonidVolohov committed:
Merge remote-tracking branch 'origin/currency-fragments' into currency-fragments

LeonidVolohov committed:
Currency-Fragment:
- Changed size of margin in fragment_currency.xml

kramarov-evg commented on app/src/main/java/com/stonks/api/ApiService.kt:
```diff
@@ -2,22 +2,31 @@ package com.stonks.api
-    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
-    .build()
-    .create(CurrencyRequests::class.java)
+        .baseUrl(CURRENCY_API_BASE_URL)
+        .addConverterFactory(GsonConverterFactory.create())
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .build()
+        .create(CurrencyRequests::class.java)
```
Why extra-indented? Code-style

LeonidVolohov replied:
Android Studio does it automatically when "Reformat code" selected during commits

kramarov-evg commented on app/src/main/java/com/stonks/api/ApiService.kt:
```diff
@@ -2,22 +2,31 @@ package com.stonks.api
-    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
-    .build()
-    .create(CryptoCurrencyRequests::class.java)
+        .baseUrl(CRYPTOCURRENCY_API_BASE_URL)
+        .addConverterFactory(GsonConverterFactory.create())
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .build()
+        .create(CryptoCurrencyRequests::class.java)
+
+val stocksApi = Retrofit.Builder()
+        .baseUrl(STOCK_API_BASE_URL)
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+        .create(StocksRequests::class.java)
```
Same about extra-indentation here

LeonidVolohov replied:
Wrote the message above about it

kramarov-evg commented on app/src/main/java/com/stonks/api/Constants.kt:
```diff
@@ -4,8 +4,9 @@ class Constants {
         const val CURRENCY_API_BASE_URL = "https://api.exchangeratesapi.io/"
 
-        const val STOCK_API_BASE_URL = "https://www.alphavantage.co/"
+        const val STOCK_API_BASE_URL = "https://www.alphavantage.co/query/"
```
No need to add "query" endpoint. All requests are anyway sent there via `@GET("endpoint")`, we've discussed it already and found it useless

kramarov-evg commented on app/src/main/java/com/stonks/api/stocks/StocksApiDataUtils.kt:
```diff
@@ -0,0 +1,187 @@
+package com.stonks.api.stocks
```
Why are you even changing anything related to stocks? Those are my files!

kramarov-evg commented on app/src/main/java/com/stonks/api/stocks/StocksDataModel.kt:
```diff
@@ -0,0 +1,38 @@
+package com.stonks.api.stocks
```
DON'T TOUCH STOCKS!!!

kramarov-evg commented on app/src/main/java/com/stonks/api/stocks/StocksRequests.kt:
```diff
@@ -0,0 +1,31 @@
+package com.stonks.api.stocks
```
STOCKS. ARE. MINE.

kramarov-evg commented on app/src/main/java/com/stonks/ui/chart/StockLineChart.kt:
```diff
@@ -40,7 +40,7 @@ class StockLineChart(lineChart: LineChart) {
     }
 
-    fun getLineData(entries: List<String>, baseCurrency: String, targetCurrency: String): LineDataSet {
+    fun getLineData(entries: List<Double>, baseCurrency: String, targetCurrency: String): LineDataSet {
```
Thanks here :+1:

kramarov-evg commented on app/src/main/java/com/stonks/ui/currency/CurrencyFragment.kt:
```diff
@@ -0,0 +1,193 @@
+            if (isChecked) {
+                when (currency_button_group.checkedButtonId) {
+                    R.id.currency_togglebutton_one_week_selector -> {
+                        currencyFragmentUtils.plotRatesPerPeriod(
+                                startDate = "1W",
+                                targetRate = targetRateSpinnerString,
+                                baseRate = baseRateSpinnerString,
+                                stockLineChart = currencyLineChart,
+                                currencyChart = currency_chart
+                        )
```
Add 1D button back. We MUST have it according to TZ. We just can say, that we have no data for the period selected (like a toast, or any other way), but we MUST provide an opportunity to select 1 day

LeonidVolohov replied:
Added

kramarov-evg commented on app/src/main/java/com/stonks/ui/currency/CurrencyFragmentUtils.kt:
```diff
@@ -0,0 +1,216 @@
+    private val TAG = CurrencyFragmentUtils::class.java.name
+    private var localDisposable = disposable
+
+    /**
+     * Умножает две строки [firstNumber]  и [secondNumber] и возвращает только первые 7 символов
+     */
```
If you're documenting functions it's a good idea to specify what are their params and return values. Not really a deadly sin here, but just for future & to refactor later

LeonidVolohov replied:
According to [this](https://stackoverflow.com/questions/38015307/generating-kotlin-method-class-comments/51067319#51067319) SO answer it is better to comment with this style

kramarov-evg commented on app/src/main/java/com/stonks/ui/main/MainActivity.kt:
```diff
@@ -37,7 +41,4 @@ class MainActivity : AppCompatActivity() {
 
             true
         }
-
-    // Variable storing currently selected fragment for displaying
-    private var selectedFragment : Fragment = DefaultFragment()  // TODO: place fragment corresponding to most left tab (now crypto)
```
Why did you delete this and not replaced `DefaultFragment()` with `CurrencyFragment()`?

LeonidVolohov replied:
It is not deleted just replaced to the top of the class

kramarov-evg commented on app/src/main/java/com/stonks/ui/stocks/StocksFragment.kt:
```diff
@@ -1,14 +1,165 @@
 package com.stonks.ui.stocks
```
Keep stocks out of your commits!!!

kramarov-evg reviewed (changes requested):
In general nice, but some things have to be fixed. Also some for-future comments. See it

LeonidVolohov committed:
Currency-Fragment:
- Removed stock files

LeonidVolohov committed:
Currency-Fragment:
- Added stocks files from develop branch

LeonidVolohov committed:
Currency-Fragment:
- Added stocks files from develop branch

LeonidVolohov committed:
Currency-Fragment:
- Created funtion in CurrencyFragment.kt for displaying chart when toggle button pressed

LeonidVolohov committed:
Merge remote-tracking branch 'origin/currency-fragments' into currency-fragments

kramarov-evg committed:
Merge branch 'develop' into currency-fragments
